### ISSUE TITLE
what is wrong happened?

### ISSUE THREAD

**Issue by vinnitu** [MajorTal/DeepSpell#9]
I have loss 0.05 at 165 iteration

```
--------------------------------------------------
Iteration 165
Train on 1580553 samples, validate on 175618 samples
Epoch 1/1
1580553/1580553 [==============================] - 2413s - loss: 0.0580 - acc: 0.9845 - val_loss: 0.2363 - val_acc: 0.9613
Q rubber bullet fired by police...........
A rubber bullet fired by police...........
☑ rubber bullet fired by police...........
---
Q other members of Congress campaigning...
A other members of Congress campaigning...
☑ other members of Congress campaigning...
---
Q The Lakers were looking to break a......
A The Lakers were looking to break a......
☑ The Lakers were looking to break a......
---
Q Francisco/New York Times as a quaint....
A Francisco/New York Times as a quaint....
☑ Francisco/New York Times as a quaint....
---
Q final night of the season...............
A final night of the season...............
☑ final night of the season...............
---
Q already half-completed 2011 fiscal year.
A already half-completed 2011 fiscal year.
☑ already half-completed 2011 fiscal year.
---
Q governments Run-After-The-Smugglers.....
A government's Run-After-The-Smugglers....
☒ governments Run-After-The-Smugglers.....
---
Q demonstrations in Cairo and several.....
A demonstrations in Cairo and several.....
☑ demonstrations in Cairo and several.....
---
Q military action, even in limited........
A military action, even in limited........
☑ military action, even in limited........
---
Q healthcare."............................
A healthcare."............................
☑ healthcare."............................
```
but after 165 iteration I see loss 2.56

what happened?

```
--------------------------------------------------
Iteration 166
Train on 1580553 samples, validate on 175618 samples
Epoch 1/1
1580553/1580553 [==============================] - 2412s - loss: 2.5680 - acc: 0.3418 - val_loss: 2.5517 - val_acc: 0.3255
Q imported commodities, can no longer be..
A imported commodities, can no longer be..
☒ in                              ........
---
Q hesitate to take ction" should it.......
A hesitate to take action" should it......
☒ hee                        .............
---
Q When the two agets grabbed him, he......
A When the two agents grabbed him, he.....
☒ The                             ........
---
Q The omst difficult thing about  food....
A The most difficult thing about a food...
☒ The  e                         .........
---
Q sincerity...............................
A sincerity...............................
☒ so  ....................................
---
Q trot onut his old New England accent....
A trot out his old New England accent.....
☒ to                         .............
---
Q publishers and rehtailers over the sale.
A publishers and retailers over the sale..
☒ pus  e                           .......
---
Q invited down to see them, but I'm.......
A invited down to see them, but I'm.......
☒ in e                        ............
---
Q during a typically slow season for......
A during a typically slow season for......
☒ doee                       .............
---
Q the national and omIal values of the....
A the national and moral values of the....
☒ the                            .........
```

and now still loss 2.85 at 181 itteration

```
--------------------------------------------------
Iteration 181
Train on 1580553 samples, validate on 175618 samples
Epoch 1/1
1580553/1580553 [==============================] - 2412s - loss: 2.8511 - acc: 0.2510 - val_loss: 2.8094 - val_acc: 0.2583
Q Medicare scam...........................
A Medicare scam...........................
☒ tee                       ..............
---
Q pleaded guilty to one counZ each of.....
A pleaded guilty to one count each of.....
☒ to            ..........................
---
Q they cannot repay.......................
A they cannot repay.......................
☒ te                       ...............
---
Q the governmnet-run Russian State Circus.
A the government-run Russian State Circus.
☒ tee                        .............
---
Q Israeli officials now talk of a.........
A Israeli officials now talk of a.........
☒ te                      ................
---
Q "My biggest concern is whether there....
A "My biggest concern is whether there....
☒ Tee                        .............
---
Q Crapo and Coburn - all fiscal...........
A Crapo and Coburn - all fiscal...........
☒ aee                    .................
---
Q statemnt in Arabic, which she...........
A statement in Arabic, which she..........
☒ ae                                      
---
Q consiering tighter controls on animal...
A considering tighter controls on animal..
☒ tee                       ..............
---
Q FAA'so peating authority through........
A FAA's operating authority through.......
☒ tee                         ............
```

**Comment by MajorTal** [comment]
Wow, I've never seen it "forget" this badly.
That said, even at iteration 165, you see the training accuracy is at 0.9845 - while the validation accuracy is stuck at 0.9613 (pretty good!). The network is starting to overfit the input data, but the performance on the testing data (which is unseen by the network during the training process) is not improving anymore.
At somewhere around that point you need to stop the training - I have not programmed an automatic stopping mechanism for this.
